lib/services/ios-project-service.ts

```diff
@@ -1045,73 +1106,148 @@ We will now place an empty obsolete compatability white screen LauncScreen.xib f
 		return null;
 	}
 
-	private readTeamId(): string {
+	private readXCConfigSigning(): IiOSBuildConfig {
+		const result: IiOSBuildConfig = {};
 		let xcconfigFile = path.join(this.$projectData.appResourcesDirectoryPath, this.platformData.normalizedPlatformName, "build.xcconfig");
 		if (this.$fs.exists(xcconfigFile).wait()) {
 			let text = this.$fs.readText(xcconfigFile).wait();
-			let teamId: string;
 			text.split(/\r?\n/).forEach((line) => {
 				line = line.replace(/\/(\/)[^\n]*$/, "");
-				if (line.indexOf("DEVELOPMENT_TEAM") >= 0) {
-					teamId = line.split("=")[1].trim();
-					if (teamId[teamId.length - 1] === ';') {
-						teamId = teamId.slice(0, -1);
+				const read = (name: string) => {
+					if (line.indexOf(name) >= 0) {
+						let value = line.substr(line.lastIndexOf("=") + 1).trim();
+						if (value.charAt(value.length - 1) === ';') {
+							value = value.substr(0, value.length - 1).trim();
+						}
+						return value;
 					}
-				}
+					return undefined;
+				};
+				result.teamIdentifier = read("DEVELOPMENT_TEAM") || result.teamIdentifier;
+				result.codeSignIdentity = read("CODE_SIGN_IDENTITY") || result.codeSignIdentity;
+				result.mobileProvisionIdentifier = read("PROVISIONING_PROFILE[sdk=iphoneos*]") || result.mobileProvisionIdentifier;
 			});
-			if (teamId) {
-				return teamId;
-			}
 		}
-		let fileName = path.join(this.platformData.projectRoot, "teamid");
-		if (this.$fs.exists(fileName).wait()) {
-			return this.$fs.readText(fileName).wait();
+		return result;
+	}
+
+	private getProvisioningProfile(): provision.MobileProvision {
```

Comment:
this method has async operations (`.wait()` inside it's body), so it must return `IFuture<provision.MobileProvision>`